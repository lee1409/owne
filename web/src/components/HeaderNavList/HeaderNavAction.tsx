import { Button, ButtonProps } from "@material-ui/core"
import { styled } from "@material-ui/core/styles"

import { HeaderNavLabel } from "../Typography"

const StyledHeaderNavButton = styled(Button)(({ theme }) => ({
  textTransform: "none",
  margin: `0 ${theme.spacing(3)}px`,
}))

type HeaderNavActionProps = {
  label: string
} & ButtonProps

const HeaderNavAction = ({ children, ...props }: HeaderNavActionProps) => {
  return (
    <StyledHeaderNavButton {...props}>
      <HeaderNavLabel>{props.label}</HeaderNavLabel>
    </StyledHeaderNavButton>
  )
}

export default HeaderNavAction
