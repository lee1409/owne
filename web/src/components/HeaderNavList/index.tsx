import { FC } from "react"
import clsx from "clsx"
import { useLocation, useHistory } from "react-router-dom"

import HeaderNavAction from "./HeaderNavAction"
import ROUTE_NAME from "../../router/routeName"

const actions = [
  {
    label: "Menu",
    route: ROUTE_NAME.MENU,
  },
  {
    label: "Stories",
    route: ROUTE_NAME.STORIES,
  },
]

type HeaderNavListProps = {
  className?: string
}

const HeaderNavList: FC<HeaderNavListProps> = ({ className }) => {
  const classes = clsx(className)
  const location = useLocation()
  const { pathname } = location
  const history = useHistory()

  const navigateTo = (route: string) => {
    history.push(route)
  }

  return (
    <div className={classes}>
      {actions.map(action => {
        const isActive = pathname === action.route
        return (
          <HeaderNavAction
            key={action.route}
            label={action.label}
            color={isActive ? "primary" : "secondary"}
            onClick={() => navigateTo(action.route)}
          />
        )
      })}
    </div>
  )
}

export default HeaderNavList
