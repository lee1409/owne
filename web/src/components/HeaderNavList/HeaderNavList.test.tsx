import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import HeaderNavList from "."

test("HeaderNavList", () => {
  const headerNavList = render(
    <MemoryRouter>
      <HeaderNavList />
    </MemoryRouter>
  )
  expect(headerNavList).toMatchSnapshot("HeaderNavList")
})
