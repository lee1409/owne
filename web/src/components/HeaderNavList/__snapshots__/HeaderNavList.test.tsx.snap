// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavList: HeaderNavList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-1 WithStyles(ForwardRef(Button))-root-2 MuiButton-textSecondary"
          label="Explore"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Explore
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-1 WithStyles(ForwardRef(Button))-root-3 MuiButton-textSecondary"
          label="Menu"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Menu
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-1 WithStyles(ForwardRef(Button))-root-4 MuiButton-textSecondary"
          label="Stories"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Stories
            </p>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-1 WithStyles(ForwardRef(Button))-root-2 MuiButton-textSecondary"
        label="Explore"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Explore
          </p>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-1 WithStyles(ForwardRef(Button))-root-3 MuiButton-textSecondary"
        label="Menu"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Menu
          </p>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-root-1 WithStyles(ForwardRef(Button))-root-4 MuiButton-textSecondary"
        label="Stories"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Stories
          </p>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
